lib/icinga/icingaapplication.cpp

```diff
@@ -291,6 +294,35 @@ String IcingaApplication::GetNodeName(void) const
 	return ScriptGlobal::Get("NodeName");
 }
 
+void IcingaApplication::DeclarePluginDir(String pluginDir)
+{
+	if (! ScriptGlobal::Exists("PluginDir")) {
+		return ScriptGlobal::Set("PluginDir", pluginDir);
+	}
+}
+
+String IcingaApplication::GetPluginDir(void) const
+{
+	return ScriptGlobal::Get("PluginDir");
+}
+
+void IcingaApplication::DeclarePluginPath(Array pluginPath)
+{
+	if (! ScriptGlobal::Exists("PluginPath")) {
+		return ScriptGlobal::Set("PluginPath", pluginPath);
+	}
+}
+
+Value IcingaApplication::GetPluginPath(void) const
+{
+	Array::Ptr path = ScriptGlobal::Get("PluginPath");
+	String dir = GetPluginDir();
+	if (dir && ! path->Contains(dir))
```

Comment:
Should possibly use `!dir.IsEmpty()`. Also remove the space between ! and path (style guide).